### ISSUE TITLE
Package structure and code improvement

### ISSUE THREAD

**Issue by zutshianand** [enochkan/torch-metrics#6]
Hi all,

With this initiative of collating all the metrics and presenting them in Pytorch, I wanted to share my thoughts on the same.
**Referring to the code structure**, is there any reason why we are having the code structure of the entire package using the name of the metric rather than the task of deep learning?

IMHO, the import should be something like ```from torch_metrics.classification import <name of metric>``` rather than ```from torch_metrics import <name of metric>```, this is because by adding the different metrics in function methods rather than separate classes, it provides the following:
* Context to the client regarding the metric usage
* Reusability across different methods of utility functions.

We can use ```sklearn.metrics``` package as an example.

Thanks!

**Comment by enochkan** [comment]
That's a great suggestion. Do you mind coming up with a list of function methods? I'm quite busy at the moment, otherwise I can get to it in a couple days. 👍

**Comment by zutshianand** [comment]
Sure will share the overall code structure.

**Comment by enochkan** [comment]
@hassiahk @zutshianand can you guys also think of ways to make this package more popular? I'm loving the feedbacks and contributions from both of you, and would love it for more people to use it haha.

**Comment by hassiahk** [comment]
@enochkan Right now I can only think of these:
- Better README.md with examples to all the available metrics and different ways they can be used.
- Example notebooks on how to implement these metrics in real word problems like how they have it in TensorFlow. This will take a lot of time though and we can probably ignore for now.
- Add more metrics and functionalities to each metric.
- Lastly to share it among social circles and ask for feedback.

These ideas just came from top of my head, so I will share some more ideas if I come across any, by raising new issues.

**Comment by hassiahk** [comment]
Hello @enochkan and @zutshianand, we can have a structure somewhat like this:
```
torch_metrics
    |-- classification
    |   |-- accuracy.py
    |   |-- dice.py
    |   |-- f1.py
    |   |-- hinge.py
    |   |-- kldivergence.py
    |   |-- pr.py
    |   `-- squarehinge.py
    |-- object_detection
    |   |-- average_precision.py
    |   `-- precision_recall_curve.py
    |-- regression
    |   |-- logcosh.py
    |   |-- mae.py
    |   |-- meaniou.py
    |   |-- mse.py
    |   |-- rmse.py
    |   `-- rsquared.py
    `-- utils.py
```
What do you guys think?

**Comment by enochkan** [comment]
that looks great. I appreciate the effort. I can keep a running bullet list of what's implemented and what needs to be done in the Readme file.

**Comment by hassiahk** [comment]
@enochkan @zutshianand Some of the metrics cannot be classified as either `Classification` or `Regression` like `DSC (Dice Similarity Coefficient)` which is mainly used for `Image Segmentation` (For this, we can create a new folder) and some of the metrics can be used in both `Classification` and `Regression` like `KL Divergence`. What should we do in such cases? How to structure these metrics?

**Comment by zutshianand** [comment]
We can essentially have two directories for Classification and Regression. AFAIK, Image segmentation is also a classification task, hence it should ideally be in Classification. Also, I think KL Divergence is mostly used in classification tasks since we deal with probabilities and their distributions in classification tasks. So we can essentially divide most of the metrics as classification or regression.

You can add a list of metrics which you feel do not fit either in classification or regression and we can decide then.

Also, in [this](https://github.com/enochkan/torch-metrics/issues/6#issuecomment-724231191), do we really need an object-detection directory as well? Precision and recall can come under classification right? What do you think?

**Comment by hassiahk** [comment]
Sure @zutshianand, will create that list.

Based on [this](https://github.com/enochkan/torch-metrics/issues/9#issuecomment-724205766) comment by @Vijayabhaskar96, I created it like that. But surely we can modify it.

**Comment by Vijayabhaskar96** [comment]
Precision and recall formula for both classification and object detection is the same but the problem is how the TP, FP, TN, and FN are calculated. If the Precision is made to take TP and FP as input instead of directly taking target and predicted tables as input, the same function can be used to calculate both Object detection and classification, but that would make the user experience bad.

**Comment by zutshianand** [comment]
I agree here. The meaning of the attributes should ideally be agnostic to the client here. IMHO, that is a good way of designing a closed solution. To rephrase, the method definition should not be dependent upon the task at hand. Since precision and recall is a very generic metric and can be used in a lot of forms, we can have this outside of all the directories and not in any of them. This way, the import would be something like this ```from torch_metrics import precision```. This would not create any form of confusion to the client (user) in any manner. Also, we would be providing additional documentation to support how the attributes can be used for different ML tasks.

Hope this makes sense!
